src/compile-middleware.ts

```diff
@@ -84,56 +104,195 @@ export function babelCompile(forceCompile: boolean): RequestHandler {
+      const plugins = [];
+      if (options.transformES2015) {
+        plugins.push(...es2015Plugins);
+      }
+      if (transformingModule) {
+        plugins.push(...modulesPlugins);
+      }
+
+      let compiled;
+      try {
+        compiled = babelCore.transform(js, {plugins}).code;
+      } catch (e) {
+        // Leave the original script as-is. Maybe it will work...
+        // TODO Show an error in the browser console, or on the runner page.
+        console.warn(`Error compiling script in ${location}: ${e.message}`);
+        continue;
+      }
+
+      if (transformingModule) {
+        // The Babel AMD transformer output always starts with a `define` call,
+        // which registers a module but does not execute it immediately. Since
+        // we're in HTML, these are our top-level scripts, and we want them to
+        // execute immediately. Swap it out for `require` so that it does.
+        compiled = compiled.replace('define', 'require');
+
+        // Remove type="module" since this is non-module JavaScript now.
+        dom5.removeAttribute(scriptTag, 'type');
+      }
+
+      dom5.setTextContent(scriptTag, compiled);
     }
   }
+
+  if (wctScriptTag && requireScriptTag) {
+    // This looks like a Web Component Tester script, and we have converted
+    // ES modules to AMD. Converting a module to AMD means that it will not
+    // execute synchronously like an ES module would. Since WCT listens for the
+    // `DOMContentLoaded` event, this means test suites in modules will not
+    // have been registered by the time WCT starts running tests.
+    //
+    // To address this, we inject a block of JS that uses WCT's `waitFor` hook
+    // to defer running tests until our AMD modules have loaded. Do this as
+    // late as possible, before the WCT script, because users may be setting
+    // their own `waitFor` that musn't clobber ours. Likewise we must call
+    // theirs if we find it.
+    dom5.insertBefore(
+        wctScriptTag.parentNode, wctScriptTag, parse5.parseFragment(`
+<script>
+  (function() {
+    window.WCT = window.WCT || {};
+    var originalWaitFor = window.WCT.waitFor;
+    window.WCT.waitFor = function(cb) {
+      window._wctCallback = function() {
+        if (originalWaitFor) {
+          originalWaitFor(cb);
+        } else {
+          cb();
+        }
+      };
+    };
+  }());
+</script>
+`));
+
+    // Monkey patch `require` to keep track of loaded AMD modules. Note this
+    // assumes that all modules are registered before `DOMContentLoaded`, but
+    // that's an assumption WCT normally makes anyway. Do this right after
+    // RequireJS is loaded, and hence before the first module is registered.
+    dom5.insertAfter(
+        requireScriptTag.parentNode, requireScriptTag, parse5.parseFragment(`
+<script>
+  (function() {
+    var originalRequire = window.require;
+    var moduleCount = 0;
+    window.require = function(deps, factory) {
+      moduleCount++;
```

Comment:
I wonder if we should have a setTimeout here that checks if this require call's factory completed, and warns if not. That way there's at least something to go on if WCT doesn't start.